core/shared/src/main/scala/com/lynbrookrobotics/potassium/events/ContinuousEvent.scala

```diff
@@ -85,11 +86,130 @@ class ContinuousEvent(condition: => Boolean)(implicit polling: ImpulseEvent) {
     * Returns a continuous event that is an intersection of both events
     * @param event the event to intersect with the original
     */
-  def &&(event: ContinuousEvent): ContinuousEvent = {
+  def &&(event: PollingContinuousEvent): PollingContinuousEvent = {
     Signal(event.conditionValue && this.conditionValue).filter(identity)
   }
 
-  def unary_!(): ContinuousEvent = {
+  def unary_!(): PollingContinuousEvent = {
     Signal(!conditionValue).filter(identity)
   }
 }
+
+
+class ContinuousEvent(val condition: () => Boolean) {
```

Comment:
Hmm, now that we are externally telling the `ContinuousEvent` its state, having this `condition` parameter seems like an antipattern. Can you make `checkCondition` be more like `updateCondition`?

Reply:
Is this just a question of naming, or do you mean a change in code design?